src/main/java/graphql/language/AstPrinter.java

```diff
@@ -712,7 +712,16 @@ private static void printImpl(StringBuilder writer, Node node, boolean compactMo
      *
      * @param <T> the type of node
      */
-    private interface NodePrinter<T extends Node> {
+    protected interface NodePrinter<T extends Node> {
         void print(StringBuilder out, T node);
     }
+
+    /**
+     * Allow subclasses to replace a printer for a specific {@link Node}
+     * @param nodeClass the class of the {@link Node}
+     * @param nodePrinter the custom {@link NodePrinter}
+     */
+    protected void replacePrinter(Class<? extends Node> nodeClass, NodePrinter<? extends Node> nodePrinter) {
```

Comment:
can mark this either either as `Internal` or make it package protected instead of `protected`